src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -60,8 +61,8 @@ public AnalyzerFileReference(string fullPath, IAnalyzerAssemblyLoader assemblyLo
             FullPath = fullPath;
             _assemblyLoader = assemblyLoader ?? throw new ArgumentNullException(nameof(assemblyLoader));
 
-            _diagnosticAnalyzers = new Extensions<DiagnosticAnalyzer>(this, IsDiagnosticAnalyzerAttribute, GetDiagnosticsAnalyzerSupportedLanguages);
-            _generators = new Extensions<ISourceGenerator>(this, IsGeneratorAttribute, GetGeneratorsSupportedLanguages);
+            _diagnosticAnalyzers = new Extensions<DiagnosticAnalyzer>(this, IsDiagnosticAnalyzerAttribute, GetDiagnosticsAnalyzerSupportedLanguages, allowNetFramework: true);
```

Comment:
>allowNetFramework: true [](start = 149, length = 23)

Why are analyzers allowed to target .NET Framework but generators are not?

Reply:
They were never supported, we just failed to enforce that fact. Now the fact that we didn't enforce is one of the hurdles for us movign fully to .NET Core. This change is an attempt to not make the same mistake, and create the same hurdle, for source generators,